plutus-use-cases/src/Plutus/Contracts/Governance.hs

```diff
@@ -175,19 +175,19 @@ transition Params{..} State{ stateData = s, stateValue} i = case (s, i) of
                 (zip initialHolders tokenNames)
         in Just (constraints <> Constraints.mustMintValue total, State s stateValue)
 
-    (GovState law mph Nothing, ProposeChange proposal@Proposal{tokenName}) ->
-        let constraints = ownsVotingToken mph tokenName
+    (GovState law mph Nothing, ProposeChange owner proposal@Proposal{tokenName}) ->
+        let constraints = ownsVotingToken owner mph tokenName
```

Comment:
Instead of adding an `owner` parameter you should use the `ownAddress` function. I'm not even sure it is secure this way since you can pass a different value than your own address.

Reply:
We're out of the contract monad at this stage though, so `ownAddress` won't work. From a security perspective, the `owner` address is the one where we'll send the vote token back, we still use the current wallet to initiate the transaction, so it should be ok.

Reply:
Oh, I get it now. It's it only for specifying where the token should go to, which `mustSpendAtLeast` didn't check either.